pkg/controllers/job/job_controller_actions.go

```diff
@@ -253,6 +253,7 @@ func (cc *Controller) syncJob(jobInfo *apis.JobInfo, updateStatus state.UpdateSt
 
 	waitCreationGroup := sync.WaitGroup{}
 	waitCreationGroup.Add(len(podToCreate))
+	stateMutex := sync.Mutex{}
```

Comment:
move mutex into function

Reply:
`classifyAndAddUpPodBaseOnPhase` is not the only place that we need to protect.

Reply:
the other also should move to function, do not put mutex in more than on place.

Reply:
FYI：
```
 // A Mutex must not be copied after first use.

```


So this is not right, you can change to stateMutex := &sync.Mutex{}

Reply:
@hzxuzhonghu  just curious, for which reason why this is not working? Which code leads to the multiple copies of mutex instance?

Reply:
Actually, this var is used by many go routines, and if you print the address of it in the go routines, you can find they are not same.  So can not protect the critical-section using different lock.

Reply:
But the mutex instance is not used as a parameter for those go routines.

Reply:
make sense, my memory messed up.